### PR TITLE
refactor(ramp): use isBuy, isSell instead of rampType compare

### PR DESCRIPTION
**Pull request by wachunei**
## **Description**

This PR changes the comparisons `rampType === RampType.BUY` to `isBuy` and  `rampType === RampType.SELL` to `isSell` 

## **Manual testing steps**

Navigate through both Buy and Sell flows and confirm there is not changes in the UI/logic

## **Screenshots/Recordings**
n/a

## **Related issues**

n/a

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by georgeweiler — APPROVED** [review]

**Comment by bkirb** [comment]
@wachunei LGTM I verified the buy and sell flows ✅

File: app/components/UI/Ramp/buy/Views/Checkout.tsx [hunk 1]
```diff
@@ -115,10 +105,9 @@ const CheckoutWebView = () => {
           return;
         }
         const orders = await SDK.orders();
-        const getOrderFromCallbackMethod =
-          rampType === RampType.BUY
-            ? 'getOrderFromCallback'
-            : 'getSellOrderFromCallback';
+        const getOrderFromCallbackMethod = isBuy
+          ? 'getOrderFromCallback'
```
Comment (tommasini) [inline app/components/UI/Ramp/buy/Views/Checkout.tsx:109]: NIT for future refactoring:  would change the name to `getBuyOrderFromCallback`

File: app/components/UI/Ramp/buy/hooks/useLimits.ts [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
     selectedPaymentMethodId,
     selectedAsset,
     selectedFiatCurrencyId,
-    rampType,
+    isBuy,
   } = useRampSDK();
 
   const [{ data: limits }] = useSDKMethod(
-    rampType === RampType.BUY ? 'getLimits' : 'getSellLimits',
+    isBuy ? 'getLimits' : 'getSellLimits',
```
Comment (tommasini) [inline app/components/UI/Ramp/buy/hooks/useLimits.ts:14]: Ditto: would change `getLimites` to `getBuyLimits`

File: app/components/UI/Ramp/buy/hooks/usePaymentMethods.ts [hunk 3]
```diff
@@ -31,10 +31,9 @@ function usePaymentMethods() {
           if (!method.customAction) {
             allowed.push(method.id);
           } else {
-            const cryptoCurrenciesMethod =
-              rampType === RampType.BUY
-                ? 'getCryptoCurrencies'
-                : 'getSellCryptoCurrencies';
+            const cryptoCurrenciesMethod = isBuy
+              ? 'getCryptoCurrencies'
```
Comment (tommasini) [inline app/components/UI/Ramp/buy/hooks/usePaymentMethods.ts:35]: ditto

File: app/components/UI/Ramp/buy/hooks/useQuotes.ts [hunk 4]
```diff
@@ -9,10 +8,10 @@ function useQuotes(amount: number) {
     selectedAsset,
     selectedAddress,
     selectedFiatCurrencyId,
-    rampType,
+    isBuy,
   } = useRampSDK();
   const [{ data, isFetching, error }, query] = useSDKMethod(
-    rampType === RampType.BUY ? 'getQuotes' : 'getSellQuotes',
+    isBuy ? 'getQuotes' : 'getSellQuotes',
```
Comment (tommasini) [inline app/components/UI/Ramp/buy/hooks/useQuotes.ts:14]: ditto

**Review by tommasini — COMMENTED** [review]
LGTM! Would change some variables naming to more consistency between sell and buy

File: app/components/UI/Ramp/buy/hooks/useLimits.ts (hunk 2 above)
Comment (wachunei) [reply app/components/UI/Ramp/buy/hooks/useLimits.ts:14]: These names are not part of the mobile project scope, these are SDK methods names https://socket.dev/npm/package/@consensys/on-ramp-sdk/files/1.25.2/dist/regions/RegionsService.d.ts#L48

What we are doing here is something like `(new SDK()).regions().getLimits()` or `(new SDK()).regions().getSellLimits()` (not actual code).

File: app/components/UI/Ramp/buy/hooks/usePaymentMethods.ts [hunk 5]
```diff
@@ -10,13 +9,14 @@ function usePaymentMethods() {
...
-    rampType,
+    isBuy,
   } = useRampSDK();
   const [isFilterLoading, setIsFilterLoading] = useState(true);
   const [allowedMethodIds, setAllowedMethodIds] = useState<string[]>();
 
-  const paymentMethodsMethod =
-    rampType === RampType.BUY ? 'getPaymentMethods' : 'getSellPaymentMethods';
+  const paymentMethodsMethod = isBuy
+    ? 'getPaymentMethods'
```
[the remark this answers is not in the document]
Comment (wachunei) [reply app/components/UI/Ramp/buy/hooks/usePaymentMethods.ts:18]: Refer to previous comment https://github.com/MetaMask/metamask-mobile/pull/7495#discussion_r1381953099

File: app/components/UI/Ramp/buy/hooks/usePaymentMethods.ts (hunk 3 above)
Comment (wachunei) [reply app/components/UI/Ramp/buy/hooks/usePaymentMethods.ts:35]: Refer to previous comment https://github.com/MetaMask/metamask-mobile/pull/7495#discussion_r1381953099

File: app/components/UI/Ramp/buy/hooks/useQuotes.ts (hunk 4 above)
Comment (wachunei) [reply app/components/UI/Ramp/buy/hooks/useQuotes.ts:14]: Refer to previous comment https://github.com/MetaMask/metamask-mobile/pull/7495#discussion_r1381953099

File: app/components/UI/Ramp/buy/Views/Checkout.tsx (hunk 1 above)
Comment (wachunei) [reply app/components/UI/Ramp/buy/Views/Checkout.tsx:109]: Refer to previous comment https://github.com/MetaMask/metamask-mobile/pull/7495#discussion_r1381953099

File: app/components/UI/Ramp/buy/hooks/useLimits.ts (hunk 2 above)
Comment (tommasini) [reply app/components/UI/Ramp/buy/hooks/useLimits.ts:14]: Thanks for explaining!

**Pull request merged**